rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -802,6 +802,7 @@ trait PromiseGeneration_buildNodeConfigurations extends PromiseGenerationService
                                    (name, p)
                                  }
                                }
+            agent           <- Box(context.nodeInfo.agentsName.headOption) ?~! "Should not be Empty"
```

Comment:
"Should not be Empty" => "The agent type is not defined. Please, report it, it is most likelly a bug".